shoes-core/lib/shoes/download.rb

```diff
@@ -140,9 +142,23 @@ def save_to_file(file_path)
       open(file_path, 'wb') { |fw| fw.print @response.body }
     end
 
-    def download_started(content_length)
-      @content_length = content_length
-      @started        = true
+    def download_started(response)
+      unless @started
+        @started = true
+        @content_length = read_content_length(response)
+
+        @response.status = [response.code, response.message]
+        response.each_header do |key|
+          @response.headers[key] = response[key]
+        end
+
+        mark_progress
```

Comment:
This is another concession to compatibility, although one that I could be convinced into changing. `open-uri` invokes the progress proc once immediately when thing are starting up.

We had specs directly checking that. Open question whether that's just over-testing (since I wrote it, I suspect it might be), but it was reason enough for me to just match it to start with.

What do you think? Would we be happy with the slightly different progress triggering order of just doing the later progress calls?